src/components/EmailNotifications/EmailNotificationsManager.js

```diff
@@ -651,35 +651,60 @@ const EmailNotificationsManager = React.memo(
   }
 )
 
-const WrappedContainer = React.memo(function AnimatedModal({ ...props }) {
+const WrappedContainer = React.memo(function WrappedContainer({ ...props }) {
   const [insideModal] = useInside('NotificationsModal')
   const { screenId } = props
 
-  if (insideModal || screenId === NOTIFICATIONS_PREFERENCES_SCREEN) {
-    return <AnimatedContainer {...props} />
-  }
+  const Container =
+    insideModal || screenId === NOTIFICATIONS_PREFERENCES_SCREEN
+      ? AnimatedContainer
+      : AnimatedBox
+
+  return <Container {...props} />
+})
+
+const AnimatedBox = ({ ...props }) => {
+  const ref = useRef(null)
+  const [height, setHeight] = useState(0)
+
+  const setRef = useCallback(node => {
```

Comment:
Not sure if I understand this; do we even need to set the ref? Should we also listen to the `node`'s height changing?

Reply:
Ah yeah i think @rperez89 already did this but we can remove `useRef` here, as it's not necessary and we can change `setRef` to be `ref`.

Also, now that you mentioned to listen to the node height we can probably create a custom hook to encapsulate the logic.

```
function useNodeHeight() {
  const [height, setHeight] = useState(0)
  const nodeObserver = useRef(null)

  const customRef = useCallback(node => {
    if (nodeObserver.current) {
      nodeObserver.current.disconnect()
    }

    if (node) {
      setHeight(node.clientHeight)
      nodeObserver.current = new ResizeObserver(entries => {
        const { height } = entries[0].contentRect
        setHeight(height)
      })

      nodeObserver.current.observe(node)
    }
  }, [])

  return [height, customRef]
}

```

Reply:
Might be something useful to consider adding to aragonUI cc @bpierre!

Reply:
Nice, yes!